jwql/instrument_monitors/nircam_monitors/claw_monitor.py

```diff
@@ -0,0 +1,394 @@
+
+    def query_mast(self):
+        """Query MAST for new NIRCam full-frame imaging data.
+
+        Returns
+        -------
+        t : astropy.table.table.Table
+            A table summarizing the new nircam imaging data.
+        """
+
+        server = "https://mast.stsci.edu"
+        JwstObs = Mast()
+        JwstObs._portal_api_connection.MAST_REQUEST_URL = server + "/portal_jwst/Mashup/Mashup.asmx/invoke"
+        JwstObs._portal_api_connection.MAST_DOWNLOAD_URL = server + "/jwst/api/v0.1/download/file"
+        JwstObs._portal_api_connection.COLUMNS_CONFIG_URL = server + "/portal_jwst/Mashup/Mashup.asmx/columnsconfig"
+        JwstObs._portal_api_connection.MAST_BUNDLE_URL = server + "/jwst/api/v0.1/download/bundle"
+        service = 'Mast.Jwst.Filtered.Nircam'
+        FIELDS = ['filename', 'program', 'observtn', 'category', 'instrume', 'productLevel', 'filter',
+                  'pupil', 'subarray', 'detector', 'datamodl', 'date_beg_mjd', 'effexptm']
+        params = {"columns": ",".join(FIELDS),
+                  "filters": [{"paramName": "pupil", "values": ['CLEAR', 'F162M', 'F164N', 'F323N', 'F405N', 'F466N', 'F470N']},
+                              {"paramName": "exp_type", "values": ['NRC_IMAGE']},
+                              {"paramName": "datamodl", "values": ['ImageModel']},  # exclude calints, which are cubemodel
+                              {"paramName": "productLevel", "values": ['2b']},  # i.e. cal.fits
+                              {"paramName": "subarray", "values": ['FULL']}, ]
+                  }
+        t = JwstObs.service_request(service, params)
+        t = t[(t['date_beg_mjd'] > self.query_start_mjd) & (t['date_beg_mjd'] < self.query_end_mjd)]
+        t.sort('date_beg_mjd')
+        filetypes = np.array([row['filename'].split('_')[-1].replace('.fits', '') for row in t])
+        t = t[filetypes == 'cal']  # only want cal.fits files, no e.g. i2d.fits
+
+        return t
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method.  See module docstrings for further details."""
+
+        logging.info('Begin logging for claw_monitor')
+
+        # Query MAST for new NIRCam full-frame imaging data from the last 2 days
+        self.query_end_mjd = Time.now().mjd
+        self.query_start_mjd = self.query_end_mjd - 2
+        self.query_start_mjd, self.query_end_mjd = 59715.28951771492, 59715.29771992559  # todo remove
+        t = self.query_mast()
+        logging.info('{} files found between {} and {}.'.format(len(t), self.query_start_mjd, self.query_end_mjd))
+
+        # Create observation-level median stacks for each filter/pupil combo, in pixel-space
+        combos = np.array(['{}_{}_{}_{}'.format(str(row['program']), row['observtn'], row['filter'], row['pupil']).lower() for row in t])
+        t['combos'] = combos
+        monitor_run = False
+        for combo in np.unique(combos):
+            tt = t[t['combos'] == combo]
```

Comment:
Would make the code more readable to rename 'tt' to something more descriptive.

Reply:
switch to mast_table_combo